### PR TITLE
Drive mode

### PR DESCRIPTION
**Pull request by imisaacwu**

### CODE REVIEW

File: src/store/middleware/inputMiddleware.js [hunk 1]
```diff
@@ -58,18 +58,34 @@ const inputMiddleware = store => next => action => {
...
-    const { tankLeft: prevTankLeft, tankRight: prevTankRight } = prevComputedInput.drive;
-    const { tankLeft, tankRight } = computedInput.drive;
-    if (tankLeft !== prevTankLeft || tankRight !== prevTankRight)
-      dispatch(requestTankDrive({ left: tankLeft, right: tankRight }));
-  } else {
-    const { straight: prevStraight, steer: prevSteer } = prevComputedInput.drive;
-    const { straight, steer } = computedInput.drive;
-    if (straight !== prevStraight || steer !== prevSteer) {
-      dispatch(requestDrive({ straight, steer }));
+  if(computedInput.drive.type === "normal") {
```
Comment (Geeoon) [inline src/store/middleware/inputMiddleware.js:61]: make sure theres a space between the ifs and the parenthesis

File: src/store/middleware/driveMiddleware.js [hunk 2]
```diff
@@ -37,6 +37,33 @@ const driveMiddleware = store => next => action => {
...
+    case requestTurnInPlaceDrive.type: {
+      if (selectMotorsAreEnabled(store.getState())) {
+        const { steer } = action.payload;
+        store.dispatch(messageRover({
+          message: {
+            type: "turnInPlaceDriveRequest",
+            steer
+          }
+        }));
+        break;
```
Comment (Geeoon) [inline src/store/middleware/driveMiddleware.js:49]: You need to move this break out of the if scope into the case scope. If the selectMotorsAreEnabled is false, then it won't break, and the next will be ran

File: src/store/middleware/driveMiddleware.js [hunk 3]
```diff
@@ -37,6 +37,33 @@ const driveMiddleware = store => next => action => {
...
+      if (selectMotorsAreEnabled(store.getState())) {
+        const { crab, steer } = action.payload;
+        store.dispatch(messageRover({
+          message: {
+            type: "crabDriveRequest",
+            crab,
+            steer
+          }
+        }));
+        break;
```
Comment (Geeoon) [inline src/store/middleware/driveMiddleware.js:63]: Same here

File: src/store/inputSlice.js [hunk 4]
```diff
@@ -146,27 +148,46 @@ function computeDriveInput(state, action) {
...
-    action.payload.pressed
+  // Spacebar or the Y button toggles tank drive if swerve mode is normal.
+  if ((action.type === keyPressed.type && action.payload.key === " ") ||
+    (action.type === gamepadButtonChanged.type &&
+      action.payload.gamepadName === "driveGamepad" &&
+      action.payload.buttonName === "Y" &&
+      action.payload.pressed)
   ) {
-    driveInput.tank = !driveInput.tank;
+    if(driveInput.type === "normal") {
```
Comment (Geeoon) [inline src/store/inputSlice.js:158]: add a space between the if and the parenthesis

File: src/components/armDexterity/SwerveDriveMode.jsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  const changeOverride = (event) => {
+    dispatch(requestSwerveDriveMode({
+      mode: mode,
+      override: event.target.checked
+    }));
+  }
+
+  return (
+    <div className="swerveDriveMode">
+      <button disabled className="rover">
```
Comment (Geeoon) [inline src/components/armDexterity/SwerveDriveMode.jsx:26]: Change this to something that isn't a button

**Review by abhaybd — APPROVED** [review]
Tested and works great, nice job!

**Pull request merged**